Saule/ObjectExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Saule
+{
+    internal static class ObjectExtensions
+    {
+        public static object GetValueOfProperty(this object obj, string propertyName)
+        {
+            if (obj == null || propertyName == null)
+            {
+                return null;
+            }
+
+            var propertyInfo = obj.GetType().GetProperty(propertyName);
+            if (propertyInfo != null)
+            {
+                return propertyInfo.GetValue(obj);
+            }
+
+            var fieldInfo = obj.GetType().GetField(propertyName);
+            if (fieldInfo != null)
```

Comment:
I'm not sure we should be reading fields. Unless this is actually used somewhere, I'd prefer we remove it.